bbq/vm/native_functions.go

```diff
@@ -286,3 +288,95 @@ var commonBuiltinTypeBoundFunctions = []NativeFunctionValue{
 }
 
 var IndexedCommonBuiltinTypeBoundFunctions = map[string]NativeFunctionValue{}
+
+func registerAllSaturatingArithmeticFunctions() {
+	for _, ty := range common.Concat(
+		sema.AllUnsignedFixedPointTypes,
+		sema.AllSignedFixedPointTypes,
+		sema.AllUnsignedIntegerTypes,
+		sema.AllSignedIntegerTypes,
+	) {
+		registerSaturatingArithmeticFunctions(ty.(sema.SaturatingArithmeticType))
+	}
+}
+
+func registerSaturatingArithmeticFunctions(t sema.SaturatingArithmeticType) {
+
+	register := func(
+		functionName string,
+		op func(context *Context, v, other interpreter.NumberValue) interpreter.NumberValue,
+	) {
+		RegisterTypeBoundFunction(
+			string(t.ID()),
```

Comment:
Just a nit: Maybe use `commons.TypeQualifier(t)` to get the type name, to ensure it is the same across the vm and compiler

Reply:
Good point! Switch to using `commons.TypeQualifier` in 0b57696242cbbd26fda8fd8db97388827830af58